demos/using-dev-build/webpack.config.js

```diff
@@ -0,0 +1,93 @@
+const path = require("path");
+const webpack = require("webpack");
+const HtmlWebpackPlugin = require("html-webpack-plugin");
+const TerserPlugin = require("terser-webpack-plugin");
+const CopyPlugin = require("copy-webpack-plugin");
+
+let canisters;
+
+function initCanisterIds() {
+  if ((dfx_network = process.env.DFX_NETWORK)) {
+    network = dfx_network;
+    console.log(`network was inferred from environment variable DFX_NETWORK`);
+  } else {
+    network = process.env.NODE_ENV === "production" ? "ic" : "local";
+    console.log(
+      `environment variable DFX_NETWORK not set, inferring network from node environment`
+    );
+  }
+
+  network_alphanum = network.replace(/[^a-zA-Z0-9]/g, "_"); // replace non-alphanumeric like dfx
+  console.log(`network is '${network}' (${network_alphanum})`);
+
+  function getCanisterIds(path) {
+    try {
+      return require(path);
+    } catch (error) {
+      console.log(
+        `No canister_ids.json found for network ${network} (${path}), try a different network..`
+      );
+      throw error;
+    }
+  }
+
+  canisters =
+    network === "ic"
+      ? getCanisterIds(path.resolve("canister_ids.json"))
+      : getCanisterIds(
+          path.resolve(".dfx", network_alphanum, "canister_ids.json")
+        );
+
+  for (const canister in canisters) {
+    process.env[canister.toUpperCase() + "_CANISTER_ID"] =
+      canisters[canister][network_alphanum];
+  }
+}
+initCanisterIds();
+
+const index_html = path.join(__dirname, path.join("webapp", "index.html"));
+const index_js = path.join(__dirname, path.join("webapp", "index.js"));
+
+module.exports = {
+  entry: { index: index_js },
+
+  mode: "production",
+
+  output: {
+    filename: "index.js",
+    path: path.join(__dirname, "dist"),
+  },
+
+  plugins: [
+    // This loads index.html as template, and embeds index.js
+    new HtmlWebpackPlugin({
+      template: index_html,
+      cache: false,
+    }),
+
+    new webpack.EnvironmentPlugin({
+      WHOAMI_CANISTER_ID: process.env.WHOAMI_CANISTER_ID,
+      II_CANISTER_ID: process.env.INTERNET_IDENTITY_CANISTER_ID,
+      DFX_NETWORK: process.env.DFX_NETWORK || "local",
+    }),
+    new webpack.ProvidePlugin({
+      Buffer: [require.resolve("buffer/"), "Buffer"],
+      process: require.resolve("process/browser"),
+    }),
+  ],
+
+  // proxy /api to port 8000 during development
+  devServer: {
+    proxy: {
```

Comment:
This part is worth mentioning in the README. That `agent-js` does a call to the same domain, yet, you want to redirect it to the local replica on 8000. Therefore, this is added.

Reply:
How would you phrase this in the README, and where would you mention it?

Reply:
Maybe add a section in `## Helpers`.

```
### Dev server

...

### Proxy server

...

### Tests

...

```

This way you could add more details for each. Although you might want to put together `npm run start` and `npm run proxy`.

Another idea, is to not mention `npm run proxy` here. Leave it as an implementation detail of the e2e test. For people that want to go deeper, they will find out. For people just getting started, giving two ways of doing the same thing could be confusing. But maybe I am too used to beginners...

Reply:
Ok, I think you have something specific in mind. How about you submit a follow up PR?